### PR TITLE
Add documentation for --experimental-debug-memory-usage

### PR DESCRIPTION
**Pull request by mknichel**

### CODE REVIEW

File: docs/02-app/01-building-your-application/06-optimizing/13-memory-usage.mdx [hunk 1]
```diff
@@ -13,6 +13,12 @@ Applications with a large amount of dependencies will use more memory.
 
 The [Bundle Analyzer](/docs/app/building-your-application/optimizing/bundle-analyzer) can help you investigate large dependencies in your application that may be able to be removed to improve performance and memory usage.
 
+## Run `next build` with `--experimental-debug-memory-usage`
+
+Starting in `14.2.0`, you can run `next build --experimental-debug-memory-usage` to run the build in a mode where Next.js
+will print out information about memory usage continuously throughout the build, such as heap usage and garbage collection
+statistics. Heap snapshots will also be taken automatically when memory usage gets close to the configured limit.
```
Comment (ijjk) [inline docs/02-app/01-building-your-application/06-optimizing/13-memory-usage.mdx:20]: Should we add a note about the `webpackBuildWorker` caveat?

File: docs/02-app/01-building-your-application/06-optimizing/13-memory-usage.mdx [hunk 2]
```diff
@@ -32,9 +38,13 @@ In Chrome DevTools, you can open the Memory tab and click on the "Load Profile"
 You can use an inspector tool to analyze the memory usage of the application.
 
 When running the `next build` or `next dev` command, add `NODE_OPTIONS=--inspect` to the beginning of the command. This will expose the inspector agent on the default port.
-If you wish to break before any user code starts, you can pass `--inspect-brk` instead.
+If you wish to break before any user code starts, you can pass `--inspect-brk` instead. While the process is running,
+you can use a tool such as Chrome DevTools to connect to the debugging port to record and analyze a snapshot of the heap to see what memory is being retained.
```
Comment (delbaoliveira) [inline docs/02-app/01-building-your-application/06-optimizing/13-memory-usage.mdx:42]:
```suggestion
If you wish to break before any user code starts, you can pass `--inspect-brk` instead. While the process is running, you can use a tool such as Chrome DevTools to connect to the debugging port to record and analyze a snapshot of the heap to see what memory is being retained.
```

**Review by delbaoliveira — APPROVED** [review]
🚀

File: docs/02-app/01-building-your-application/06-optimizing/13-memory-usage.mdx (hunk 1 above)
Comment (mknichel) [reply docs/02-app/01-building-your-application/06-optimizing/13-memory-usage.mdx]: Added.

File: docs/02-app/01-building-your-application/06-optimizing/13-memory-usage.mdx [hunk 3]
```diff
@@ -13,6 +13,12 @@ Applications with a large amount of dependencies will use more memory.
 
 The [Bundle Analyzer](/docs/app/building-your-application/optimizing/bundle-analyzer) can help you investigate large dependencies in your application that may be able to be removed to improve performance and memory usage.
 
+## Run `next build` with `--experimental-debug-memory-usage`
+
+Starting in `14.2.0`, you can run `next build --experimental-debug-memory-usage` to run the build in a mode where Next.js will print out information about memory usage continuously throughout the build, such as heap usage and garbage collection statistics. Heap snapshots will also be taken automatically when memory usage gets close to the configured limit.
+
+> **Good to know**: This feature is not compatible with the Webpack build worker option.
```
Comment (ijjk) [inline docs/02-app/01-building-your-application/06-optimizing/13-memory-usage.mdx:20]:
```suggestion
> **Good to know**: This feature is not compatible with the Webpack build worker option which is auto-enabled unless you have custom webpack config.
```

**Review by ijjk — APPROVED** [review]

**Pull request merged**